ibis/sql/postgres/udf/api.py

```diff
@@ -90,6 +92,8 @@ def existing_udf(name, input_types, output_type, schema=None, parameters=None):
             ).format(len(input_types), len(parameters))
         )
 
+    v.validate_output_type(output_type)
```

Comment:
postgres udf doesn't necessarily need to follow the restriction of `ibis.udf.vectorized` we should not change anything about postgres udf in this PR

Reply:
Postgres the some other backends also have UDFs with the pattern of having an `input_type` list and single `output_type` so these changes were to do validation for those too (basically, I was expanding the PR to apply to all UDFs in general with this change not just vectorized UDFs)

Reply:
Let's limit the scope to Pandas and Pyspark for now.

Reply:
i am ok with validating everything, that in fact is the premise that we have the same udf signatures across backends, which AFAICT is true.

Reply:
> Let's limit the scope to Pandas and Pyspark for now.

This made sense to me (adding validation only to vectorized UDFs) especially since if the validation should be done in the node using rlz, it easily can't be applied to UDFs in other backends

But I've started to think it is better to have the validation outside of the node (basically, where the validation checks are now in this PR) (see https://github.com/ibis-project/ibis/pull/2198#discussion_r433911429)

In that case, UDFs in other backends can be validated pretty easily and using the same function as we use for vectorized UDFs, `validate_output_type` (the current state of this PR does this already, so validating everything would just mean leaving the changes in the other backends as-is instead of reverting them)

I feel like it makes sense to leave in the validation for all UDFs in this PR. what do you think?

Reply:
> i am ok with validating everything, that in fact is the premise that we have the same udf signatures across backends, which AFAICT is true.

From what I saw most of the backends have UDFs that have signatures that are pretty similar (but not exactly the same) (bigquery, impala, and spark have `input_type` and `output_type`). [SQLite's UDF](https://github.com/ibis-project/ibis/blob/master/ibis/sql/sqlite/client.py#L28) doesn't have `output_type` though

Reply:
Ok with validating.